docs/faqs.md

```diff
@@ -0,0 +1,398 @@
+			dns googleclouddns {
+				gcp_project my-gcp-project
+			}
+		}
+	}
+
+}
+```
+
+## Can I use local or remote Terraform Modules in Coder templates?
+
+One way is to reference a Terraform module from a GitHub repo to avoid
+duplication and then just extend it or pass template-specific
+parameters/resources:
+
+```hcl
+# template1/main.tf
+module "central-coder-module" {
+  source = "github.com/yourorg/central-coder-module"
+  myparam = "custom-for-template1"
+}
+
+resource "ebs_volume` `custom_template1_only_resource ` {
+}
+```
+
+```hcl
+# template2/main.tf
+module "central-coder-module" {
+  source = "github.com/yourorg/central-coder-module"
+  myparam = "custom-for-template2"
+  myparam2 = "bar"
+}
+
+resource "aws_instance` `custom_template2_only_resource ` {
+}
+```
+
+Another way using local modules is to symlink the module directory inside the
+template directory and then `tar` the template.
+
+```sh
+ln -s modules template_1/modules
+tar -cvh -C ./template_1 | coder templates <push|create> -d - <name>
+```
+
+References:
+
+- [Public Github Issue 6117](https://github.com/coder/coder/issues/6117)
+- [Public Github Issue 5677](https://github.com/coder/coder/issues/5677)
+- [Coder docs: Templates/Change Management](https://coder.com/docs/v2/latest/templates/change-management)
+
+## Can I run Coder in an air-gapped or offline mode? (no Internet)?
+
+Yes, Coder can be deployed in air-gapped or offline mode.
+https://coder.com/docs/v2/latest/install/offline
+
+Our product bundles with the Terraform binary so assume access to terraform.io
+during installation. The docs outline rebuilding the Coder container with
+Terraform built-in as well as any required Terraform providers.
+
+Direct networking from local SSH to a Coder workspace needs a STUN server. Coder
+defaults to Google's STUN servers, so you can either create your STUN server in
+your network or disable and force all traffic through the control plane's DERP
+proxy.
+
+## Create a randomized computer_name for an Azure VM
+
+Azure VMs have a 15 character limit for the `computer_name` which can lead to
+duplicate name errors.
+
+This code produces a hashed value that will be difficult to replicate.
+
+```hcl
+locals {
+concatenated_string = "${data.coder_workspace.me.name}+${data.coder_workspace.me.owner}"
+hashed_string = md5(local.concatenated_string)
+truncated_hash = substr(local.hashed_string, 0, 16)
+}
+```
+
+## Do you have example JetBrains Gateway templates?
+
+In August 2023, JetBrains certified the Coder plugin signifying enhanced
+stability and reliability.
+
+The Coder plugin will appear in the Gateway UI when opened.
+
+Selecting the most suitable template depends on how the deployment manages
+JetBrains IDE versions. If downloading from
+[jetbrains.com](https://www.jetbrains.com/remote-development/gateway/) is
+acceptable, see the example templates below which specifies the product code,
+IDE version and build number in the `coder_app` resource. This will present an
```

Comment:
```suggestion
IDE version and build number in the [`coder_app`](https://registry.terraform.io/providers/coder/coder/latest/docs/resources/app) resource. This will present an
```